command/version/command.go

```diff
@@ -73,21 +79,29 @@ type command struct {
 	cobraCommand *cobra.Command
 
 	// Settings.
-	description string
-	gitCommit   string
-	name        string
-	source      string
+	description   string
+	gitCommit     string
+	name          string
+	source        string
+	wipVersion    string
+	activeVersion string
 }
 
 func (c *command) CobraCommand() *cobra.Command {
 	return c.cobraCommand
 }
 
 func (c *command) Execute(cmd *cobra.Command, args []string) {
-	fmt.Printf("Description:    %s\n", c.description)
-	fmt.Printf("Git Commit:     %s\n", c.gitCommit)
-	fmt.Printf("Go Version:     %s\n", runtime.Version())
-	fmt.Printf("Name:           %s\n", c.name)
-	fmt.Printf("OS / Arch:      %s / %s\n", runtime.GOOS, runtime.GOARCH)
-	fmt.Printf("Source:         %s\n", c.source)
+	fmt.Printf("Description:     %s\n", c.description)
+	fmt.Printf("Git Commit:      %s\n", c.gitCommit)
+	fmt.Printf("Go Version:      %s\n", runtime.Version())
+	fmt.Printf("Name:            %s\n", c.name)
+	fmt.Printf("OS / Arch:       %s / %s\n", runtime.GOOS, runtime.GOARCH)
+	fmt.Printf("Source:          %s\n", c.source)
+	if c.wipVersion != "" {
+		fmt.Printf("WIP Version:     %s\n", c.wipVersion)
+	}
+	if c.activeVersion != "" {
+		fmt.Printf("Active Version:  %s\n", c.activeVersion)
+	}
```

Comment:
Cool you started working on this. My idea was to inject all the version bundles and print all this as yaml. We would gain a lot of this and you can then probably also use it in e2e tests how you like. At the end we should actually test all available versions and there are probably more than one at times. Boiling this down to only some WIP version or active version is not open enough IMO. Imagine we could also create Github releases with proper changelogs automatically when having all the version information available during the build.

Reply:
Sounds really well :) will push changes to try to implement it shortly.

Reply:
Done @xh3b4sd PTAL, this is how the output looks like for aws-operator:
```
description: The aws-operator handles Kubernetes clusters running on a Kubernetes
  cluster inside of AWS.
gitcommit: n/a
name: aws-operator
source: https://github.com/giantswarm/aws-operator
goversion: go1.9.3
os: linux
arch: amd64
versionbundles:
- changelogs:
  - component: calico
    description: Calico version updated.
    kind: changed
  - component: docker
    description: Docker version updated.
    kind: changed
  - component: etcd
    description: Etcd version updated.
    kind: changed
  - component: kubedns
    description: KubeDNS version updated.
    kind: changed
  - component: kubernetes
    description: Kubernetes version updated.
    kind: changed
  - component: nginx-ingress-controller
    description: Nginx-ingress-controller version updated.
    kind: changed
  components:
  - name: calico
    version: 2.6.2
  - name: docker
    version: 1.12.6
  - name: etcd
    version: 3.2.7
  - name: kubedns
    version: 1.14.5
  - name: kubernetes
    version: 1.8.1
  - name: nginx-ingress-controller
    version: 0.9.0
  dependencies: []
  deprecated: true
  name: aws-operator
  time: 2017-11-29T16:16:00Z
  version: 0.1.0
  wip: false
- changelogs:
  - component: cloudformation
    description: First version of Cloud Formation resources.
    kind: added
  components:
  - name: calico
    version: 2.6.2
  - name: docker
    version: 1.12.6
  - name: etcd
    version: 3.2.7
  - name: kubedns
    version: 1.14.5
  - name: kubernetes
    version: 1.8.1
  - name: nginx-ingress-controller
    version: 0.9.0
  dependencies: []
  deprecated: true
  name: aws-operator
  time: 2017-11-29T16:16:00Z
  version: 0.2.0
  wip: false
- changelogs:
  - component: kubernetes
    description: Updated to kubernetes 1.8.4. Fixes a goroutine leak in the k8s api.
    kind: changed
  components:
  - name: calico
    version: 2.6.2
  - name: docker
    version: 1.12.6
  - name: etcd
    version: 3.2.7
  - name: kubedns
    version: 1.14.5
  - name: kubernetes
    version: 1.8.4
  - name: nginx-ingress-controller
    version: 0.9.0
  dependencies: []
  deprecated: true
  name: aws-operator
  time: 2017-12-05T13:00:00Z
  version: 1.0.0
  wip: false
- changelogs:
  - component: CloudFormation
    description: All AWS resources use CloudFormation except KMS and S3.
    kind: added
  - component: Kubernetes
    description: Updated to Kubernetes 1.9.2.
    kind: changed
  - component: Kubernetes
    description: Switched to vanilla (previously CoreOS) hyperkube image.
    kind: changed
  - component: Docker
    description: Updated to 17.09.0-ce.
    kind: changed
  - component: Calico
    description: Updated to 3.0.1.
    kind: changed
  - component: CoreDNS
    description: Version 1.0.5 replaces kube-dns.
    kind: added
  - component: Nginx Ingress Controller
    description: Updated to 0.10.2.
    kind: changed
  components:
  - name: calico
    version: 3.0.1
  - name: docker
    version: 17.09.0
  - name: etcd
    version: 3.2.7
  - name: coredns
    version: 1.0.5
  - name: kubernetes
    version: 1.9.2
  - name: nginx-ingress-controller
    version: 0.10.2
  dependencies: []
  deprecated: true
  name: aws-operator
  time: 2018-01-22T16:00:00Z
  version: 2.0.0
  wip: false
- changelogs:
  - component: cloudconfig
    description: Fix eventual decryption race.
    kind: fixed
  components:
  - name: calico
    version: 3.0.1
  - name: docker
    version: 17.09.0
  - name: etcd
    version: 3.2.7
  - name: coredns
    version: 1.0.5
  - name: kubernetes
    version: 1.9.2
  - name: nginx-ingress-controller
    version: 0.10.2
  dependencies: []
  deprecated: true
  name: aws-operator
  time: 2018-01-30T11:55:00Z
  version: 2.0.1
  wip: false
- changelogs:
  - component: cloudformation
    description: Add missing region to host account credentials.
    kind: fixed
  components:
  - name: calico
    version: 3.0.1
  - name: docker
    version: 17.09.0
  - name: etcd
    version: 3.2.7
  - name: coredns
    version: 1.0.5
  - name: kubernetes
    version: 1.9.2
  - name: nginx-ingress-controller
    version: 0.10.2
  dependencies: []
  deprecated: false
  name: aws-operator
  time: 2018-01-31T10:43:00Z
  version: 2.0.2
  wip: false
```

Reply:
Cool. Any opinions on using YAML in favour of JSON? I said YAML before but just wondering if it would be better to work with when it would just be JSON. Uncertain.

Reply:
My 2c, YAML matches more closely what we had before and is more human-friendly, also for reusing the output down the line for further automation both are equally easy to manage (even from cli with `jq` and `yq`). In terms of implementation is just a matter of selecting the marshaler, the flag you suggest sounds interesting.